### ISSUE TITLE
Sequence * with cigar strings

### ISSUE THREAD

**Issue by jkbonfield** [enasequence/cramtools#26]
I have a SAM file (htslib/test/ce#5b.sam) that has a cigar string (7S20M1D23M10I30M10S) and sequence/quality of *.

This is legal, but is not supported from Cramtools.  I get this error:

```
Exception in thread "main" java.lang.reflect.InvocationTargetException
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:606)
        at net.sf.cram.CramTools.invoke(CramTools.java:93)
        at net.sf.cram.CramTools.main(CramTools.java:123)
Caused by: java.lang.ArrayIndexOutOfBoundsException: 14
        at net.sf.cram.Bam2Cram.updateTracks(Bam2Cram.java:93)
        at net.sf.cram.Bam2Cram.convert(Bam2Cram.java:132)
        at net.sf.cram.Bam2Cram.main(Bam2Cram.java:387)
        ... 6 more
```

**Comment by jkbonfield** [comment]
A simple example is in htslib/test/c1#unknown.seq. Or a one liner here:

```
@SQ     SN:c1   LN:10
sq1     0       c1      1       0       10M     *       0       0       *       *
```

**Comment by vadimzalunin** [comment]
fix 3b3a3a465966ba254d1d2a3e80559e113f76b2f6
